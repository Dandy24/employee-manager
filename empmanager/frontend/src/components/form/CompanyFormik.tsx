import {Formik} from "formik";
import React from "react";

export interface CompanyFormikProps{
    onSubmit: (values: any) => void
    initialName: string,
    initialPhone: number,
    initialAddress: string,
    children: any
}

export function CompanyFormik(props: CompanyFormikProps): JSX.Element {

    const { initialName, initialPhone, initialAddress, onSubmit } = props

    return(
        <Formik
            initialValues={{
                nazev: initialName ,
                //telefon: initialPhone,
                adresa: initialAddress

            }}

            onSubmit={onSubmit}>

            {props.children}

        </Formik>
    )

}
