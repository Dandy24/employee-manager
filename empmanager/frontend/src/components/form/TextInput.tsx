import {Space, Alert} from 'antd';
import { Input } from 'formik-antd';
import { useField } from 'formik';
import { SpaceSize } from 'antd/es/space';

export interface TextInputProps{
    label: string,
    spacesize: SpaceSize,
    name: string,
}

export function TextInput(props: TextInputProps): JSX.Element {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const [field, meta] = useField(props);
    const { label, spacesize, name } = props;

    return (
        <div>
            <Space size={spacesize}>
                <label htmlFor={name}>{label}</label>
                <Input name={name} />
                {meta.touched && meta.error
                    ? (
                        <Alert
                            message={meta.error}
                            type="error"
                            showIcon
                        />
                    ) : null}
            </Space>
        </div>
    );
}
