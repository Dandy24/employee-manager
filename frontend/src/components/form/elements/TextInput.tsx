import React from 'react';
import { Alert } from 'antd';
import { Input } from 'formik-antd';
import { useField } from 'formik';
import { observer } from 'mobx-react-lite';

export interface TextInputProps {
    label: string;
    name: string;
}

export const TextInput: React.FC<TextInputProps> = observer((props: TextInputProps): JSX.Element => {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const [field, meta] = useField(props);
    const { label, name } = props;

    return (
        // <InputWrapper data-testid="text-input">
        <div data-testid={`${name}-text-input`}>
            <label htmlFor={name} data-testid="text-input-label">
                {label}
            </label>
            <Input id={name} name={name} data-testid="text-input-field" />
            {meta.touched && meta.error ? (
                <Alert message={meta.error} type="error" data-testid="text-input-error" />
            ) : null}
        </div>
        // </InputWrapper>
    );
});
