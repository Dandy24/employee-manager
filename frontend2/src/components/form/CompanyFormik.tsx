import { Formik } from 'formik';
import React from 'react';
import { observer } from 'mobx-react-lite';
import { CompanyDto } from '../../models/dtos/company-dto';
import * as yup from 'yup';

export interface CompanyFormikProps {
    onSubmit: (values: CompanyDto) => void;
    initialValues: CompanyDto;
    children: React.ReactNode;
}

export const CompanyValidationSchema = yup.object({
    name: yup
        .string()
        .typeError('Spatne zadane jmeno')
        .required('Pole musí být vyplněné')
        .max(50, 'Maximalni povolena delka prekrocena'),
    phone: yup
        .string()
        .typeError('Pole musí být vyplněné')
        .required('Pole musí být vyplněné')
        .matches(/^[0-9]+$/, 'Povoleny jsou pouze číslice')
        .min(12, 'Číslo musí mít 12 číslic')
        .max(12, 'Číslo musí mít 12 číslic'),
    address: yup.string().typeError('Spatna adresa').required('Pole musí být vyplněné'),
});

export const CompanyFormik: React.FC<CompanyFormikProps> = observer((props: CompanyFormikProps): JSX.Element => {
    const { initialValues, onSubmit } = props;

    return (
        <Formik initialValues={initialValues} onSubmit={onSubmit} validationSchema={CompanyValidationSchema}>
            {props.children}
        </Formik>
    );
});
