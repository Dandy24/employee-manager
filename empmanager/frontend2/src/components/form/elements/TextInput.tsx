import React from 'react';
import { Alert } from 'antd';
import Field, { Input } from 'formik-antd';
import { useField } from 'formik';
import { SpaceSize } from 'antd/es/space';
import { InputWrapper } from '../../layout/form/InputWrapper';
import { observer } from 'mobx-react-lite';

export interface TextInputProps {
    label: string;
    spacesize: SpaceSize;
    name: string;
}

export const TextInput: React.FC<TextInputProps> = observer((props: TextInputProps): JSX.Element => {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const [field, meta] = useField(props);
    const { label, spacesize, name } = props;

    return (
        // <InputWrapper data-testid="text-input">
        <div data-testid="text-input">
            <label htmlFor={name} data-testid="text-input-label">
                {label}
            </label>
            <Input name={name} data-testid="text-input-field" />
            {meta.touched && meta.error ? (
                <Alert message={meta.error} type="error" showIcon data-testid="text-input-error" />
            ) : null}
        </div>
        // </InputWrapper>
    );
});
