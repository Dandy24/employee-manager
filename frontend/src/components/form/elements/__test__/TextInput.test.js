import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { TextInput } from '../TextInput';
import React from 'react';
import { Form, Formik } from 'formik';
import userEvent from '@testing-library/user-event';
import { CompanyValidationSchema } from '../../CompanyFormik';

test('TextInput snapshot matches the previous one', async () => {
    const { asFragment } = render(
        <Formik
            validationSchema={CompanyValidationSchema}
            initialValues={{
                name: '',
            }}
        >
            <Form>
                <TextInput label="Nazev firmy" name="name" />
            </Form>
        </Formik>,
    );

    expect(asFragment()).toMatchSnapshot();
});

test('Test correct value in text input after user types', async () => {
    render(
        <Formik
            validationSchema={CompanyValidationSchema}
            initialValues={{
                name: '',
            }}
        >
            <Form>
                <TextInput label="Nazev firmy" name="name" />
            </Form>
        </Formik>,
    );

    const textField = await screen.findByTestId('text-input-field');

    userEvent.type(await screen.findByTestId('text-input-field'), 'test');
    fireEvent.blur(await screen.findByTestId('text-input-field'));

    await waitFor(() => {
        expect(textField).toHaveDisplayValue('test');
        expect(screen.queryByTestId('text-input-error')).not.toBeInTheDocument();
    });
});

test('Test missing value validation in text input', async () => {
    render(
        <Formik
            validationSchema={CompanyValidationSchema}
            initialValues={{
                name: '',
            }}
        >
            <Form>
                <TextInput label="Nazev firmy" name="name" />
            </Form>
        </Formik>,
    );

    userEvent.click(await screen.findByTestId('text-input-field'));
    fireEvent.blur(await screen.findByTestId('text-input-field'));

    await waitFor(async () => {
        expect(await screen.findByTestId('text-input-error')).toBeInTheDocument();
        expect(await screen.findByText('Pole musí být vyplněné')).toBeInTheDocument();
    });
});

test('Test valid value validation in text input', async () => {
    render(
        <Formik
            validationSchema={CompanyValidationSchema}
            initialValues={{
                name: '',
            }}
        >
            <Form>
                <TextInput label="Nazev spolecnosti" name="name" />
            </Form>
        </Formik>,
    );

    userEvent.click(await screen.findByTestId('text-input-field'));
    userEvent.type(
        await screen.findByTestId('text-input-field'),
        'This text is longer than 50 characters.................................................',
    );
    fireEvent.blur(await screen.findByTestId('text-input-field'));

    await waitFor(async () => {
        expect(await screen.findByTestId('text-input-error')).toBeInTheDocument();
        expect(await screen.findByText('Maximální povolená délka překročena')).toBeInTheDocument();
    });
});
