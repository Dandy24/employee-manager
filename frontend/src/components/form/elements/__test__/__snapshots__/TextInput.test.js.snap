// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput snapshot matches the previous one 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      data-testid="name-text-input"
    >
      <label
        data-testid="text-input-label"
        for="name"
      >
        Nazev firmy
      </label>
      <input
        class="ant-input"
        data-testid="text-input-field"
        id="name"
        name="name"
        type="text"
        value=""
      />
    </div>
  </form>
</DocumentFragment>
`;
